src/main/80C50.c

```diff
@@ -1,39 +1,372 @@
+
+    tempX = mf->mf[0][0];
+    tempY = mf->mf[1][0];
+    mf->mf[0][0] = tempX * cos + tempY * sin;
+    mf->mf[1][0] = tempY * cos - tempX * sin;
+
+    tempX = mf->mf[0][1];
+    tempY = mf->mf[1][1];
+    mf->mf[0][1] = tempX * cos + tempY * sin;
+    mf->mf[1][1] = tempY * cos - tempX * sin;
+
+    tempX = mf->mf[0][2];
+    tempY = mf->mf[1][2];
+    mf->mf[0][2] = tempX * cos + tempY * sin;
+    mf->mf[1][2] = tempY * cos - tempX * sin;
+
+    tempX = mf->mf[0][3];
+    tempY = mf->mf[1][3];
+    mf->mf[0][3] = tempX * cos + tempY * sin;
+    mf->mf[1][3] = tempY * cos - tempX * sin;
+}
+
+void func_800805C8(MtxF* mf, f32 angle) {
+    f32 sin = sinf(angle);
+    f32 cos = cosf(angle);
+    f32 tempZ;
+    f32 tempX;
+
+    tempZ = mf->mf[2][0];
+    tempX = mf->mf[0][0];
+    mf->mf[2][0] = tempZ * cos + tempX * sin;
+    mf->mf[0][0] = tempX * cos - tempZ * sin;
+
+    tempZ = mf->mf[2][1];
+    tempX = mf->mf[0][1];
+    mf->mf[2][1] = tempZ * cos + tempX * sin;
+    mf->mf[0][1] = tempX * cos - tempZ * sin;
+
+    tempZ = mf->mf[2][2];
+    tempX = mf->mf[0][2];
+    mf->mf[2][2] = tempZ * cos + tempX * sin;
+    mf->mf[0][2] = tempX * cos - tempZ * sin;
+
+    tempZ = mf->mf[2][3];
+    tempX = mf->mf[0][3];
+    mf->mf[2][3] = tempZ * cos + tempX * sin;
+    mf->mf[0][3] = tempX * cos - tempZ * sin;
+}
+
+void func_800806B0(MtxF* mf, f32 angle) {
+    f32 sin = sinf(angle);
+    f32 cos = cosf(angle);
+    f32 tempY;
+    f32 tempZ;
+
+    tempY = mf->mf[1][0];
+    tempZ = mf->mf[2][0];
+    mf->mf[1][0] = tempY * cos + tempZ * sin;
+    mf->mf[2][0] = tempZ * cos - tempY * sin;
+
+    tempY = mf->mf[1][1];
+    tempZ = mf->mf[2][1];
+    mf->mf[1][1] = tempY * cos + tempZ * sin;
+    mf->mf[2][1] = tempZ * cos - tempY * sin;
+
+    tempY = mf->mf[1][2];
+    tempZ = mf->mf[2][2];
+    mf->mf[1][2] = tempY * cos + tempZ * sin;
+    mf->mf[2][2] = tempZ * cos - tempY * sin;
+    
+    tempY = mf->mf[1][3];
+    tempZ = mf->mf[2][3];
+    mf->mf[1][3] = tempY * cos + tempZ * sin;
+    mf->mf[2][3] = tempZ * cos - tempY * sin;
+}
+
+void func_80080798(MtxF* mf, f32 x, f32 y) {
+    mf->mf[3][0] += (mf->mf[0][0] * x) + (mf->mf[1][0] * y);
+    mf->mf[3][1] += (mf->mf[0][1] * x) + (mf->mf[1][1] * y);
+    mf->mf[3][2] += (mf->mf[0][2] * x) + (mf->mf[1][2] * y);
+    mf->mf[3][3] += (mf->mf[0][3] * x) + (mf->mf[1][3] * y);
+}
+
+void func_80080824(MtxF* mf, f32 x, f32 y, f32 z) {
+    if (x != 1.0f) {
+        mf->mf[0][0] *= x;
+        mf->mf[0][1] *= x;
+        mf->mf[0][2] *= x;
+        mf->mf[0][3] *= x;
+    }
+    if (y != 1.0f) {
+        mf->mf[1][0] *= y;
+        mf->mf[1][1] *= y;
+        mf->mf[1][2] *= y;
+        mf->mf[1][3] *= y;
+    }
+    if (z != 1.0f) {
+        mf->mf[2][0] *= z;
+        mf->mf[2][1] *= z;
+        mf->mf[2][2] *= z;
+        mf->mf[2][3] *= z;
+    }
+}
+
+void func_80080914(MtxF* mf, f32 arg1, f32 arg2, f32 arg3, f32 arg4);
 #pragma GLOBAL_ASM("asm/nonmatchings/main/80C50/func_80080914.s")
```

Comment:
Might want to just post the scratch url here as well. It might get lost in discord.